k8s-operator/api-proxy/proxy.go

```diff
@@ -283,6 +309,100 @@ func (ap *APIServerProxy) sessionForProto(w http.ResponseWriter, r *http.Request
 	ap.rp.ServeHTTP(h, r.WithContext(whoIsKey.WithValue(r.Context(), who)))
 }
 
+func (ap *APIServerProxy) recordRequestAsEvent(req *http.Request, who *apitype.WhoIsResponse) error {
+	failOpen, addrs, err := determineRecorderConfig(who)
+	if err != nil {
+		ap.log.Errorf("error trying to determine whether the kubernetes api request needs to be recorded: %v", err)
+		return err
+	}
+	if failOpen && len(addrs) == 0 {
+		return err
+	}
+
+	if !failOpen && len(addrs) == 0 {
+		ap.log.Errorf("forbidden: kubernetes api request must be recorded, but no recorders are available.")
+		return err
```

Comment:
Should we just return the "forbidden" as an error here instead? `err` will be nil from what I can see?